ui/test-file/collapseStore.ts

```diff
@@ -0,0 +1,24 @@
+// This static class keeps track of the collapse state of every describe block that is shown in the UI.
```

Comment:
Can't we use component state or context to share state than having a single store like this?

Reply:
The components get destroyed and recreated each time you switch to a different test file. One of the aspects of this feature that I was trying to go for was for it to remember which blocks were collapsed as I switched from one test file to another and back again. For instance, I might be working in one test file and have collapsed a portion of it, then I need to check out a different test file for some information and when I come back to the first test file, I would like it to have remembered the collapse state. 
My understanding of the component structure is that there is only one test-file component that keeps getting reused each time the current file is switched.  This means that the test-items (where I would store the collapsed state) is getting deleted and re-created each time we swap files. 

The collapsed state is global application state and if we were using Redux, then it would go there, but it didn't seem reasonable to introduce the overhead and complexity of Redux just for this. Not to mention that I am new to React and have no experience with Redux. I have a bit more experience at this point with Vue/Vuex. 

So the bottom line, to answer your question is that it is a store like this because I wanted the app to remember the collapsed state as I switch between files of the project.

Reply:
That makes sense. We could use the [`Context` API](https://reactjs.org/docs/context.html) to share global state rather than introducing a singleton?

Reply:
Probably. I didn't know about the context API until I ran into it last week when I learned about Vue's equivalent api. 
I will look into it.

Reply:
This is going to take much longer than I have time for right now. The Context API seems much more complex when coupled with functional components like you are using. It is pretty straight forward for a class based component. 
I will have to learn how to conver to a class based component or learn how to deal with the context in the functional component. At first glance it looks like using functional components means we would introduce another functional component whose job is soley to call the existing component, injecting the context. Seems clunky.

Reply:
I'm ok to keep this implementation for now.